File: java_test/src/DesignPattern/TemplateMethod/Barista/Tea.java
package DesignPattern.TemplateMethod.Barista;

/**
 * Created by Administrator on 2018/4/28.
 */
public class Tea extends CaffeineBeverage {
    public void brew() {
        System.out.println("Steeping the tea");
    }

    public void addCondiments() {
        System.out.println("Adding Lemon");
    }
}

File: test_go/src/pattern/09_template_method/eg2/tea.go
package eg2

import "fmt"

type Tea struct {
	baseCaffeineBeverage
}

func NewTea() *Tea {
	this := new(Tea)
	this.CaffeineBeverage = this
	return this
}

func (this *Tea) Brew() {
	fmt.Println("Steeping the tea")
}

func (this *Tea) AddCondiments() {
	fmt.Println("Adding Lemon")
}